src/lib/CombatCalc.ts

```diff
@@ -1651,13 +1670,16 @@ export default class CombatCalc {
 
     // 1. until the amount of hp values remaining above zero is more than our desired epsilon accuracy,
     //    or we reach the maximum iteration rounds
-    for (let hit = 0; hit < (TTK_DIST_MAX_ITER_ROUNDS + 1) && epsilon >= TTK_DIST_EPSILON; hit++) {
+    for (let tick = 1; tick < (TTK_DIST_MAX_ITER_ROUNDS + 1) && epsilon >= TTK_DIST_EPSILON; tick++) {
       const nextHps = new Float64Array(this.monster.skills.hp + 1);
+      const playerAttacks = !((tick - 1) % speed);
+      const thrallAttacks = this.player.buffs.thrallSpell && !((tick - 1) % THRALL_SPEED);
```

Comment:
If you're going to -1 everything everywhere you may as well keep this starting at 0. Just add one for the map get/sets.